File: android/lca/src/main/java/com/dreampany/lca/api/misc/Response.java
package com.dreampany.lca.api.misc;

/**
 * Created by Hawladar Roman on 24/6/18.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 */
public abstract class Response<T> {
    protected final T data;

    protected Response(T data) {
        this.data = data;
    }

    public T getData() {
        return data;
    }
}

File: android/tools/src/main/java/com/dreampany/tools/api/crypto/remote/response/cmc/Response.kt
package com.dreampany.tools.api.crypto.remote.response.cmc

import com.dreampany.framework.data.enums.Code

/**
 * Created by roman on 17/4/20
 * Copyright (c) 2020 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
abstract class Response<T> {
    var status: Status? = null
    var data: T? = null

    val hasError: Boolean
        get() = status?.code != Code.SUCCESS.code

    val isEmpty: Boolean
        get() {
            val raw = data ?: return true
            if (raw is List<*>) raw.isEmpty()
            if (raw is Array<*>) raw.isEmpty()
            return false
        }
}